### PR TITLE
Internal Query : Add Force Passthrough API 

### PR DESCRIPTION
**Pull request by bchong95**
# Internal Query : Add Force Passthrough API 

## Description

For compute gateway we need to honor the old legacy gateway behavior of passing through their query if a pkrangeid header is set. This is because a user might be coming from an SDK that fans out the complex queries to the gateway one partition at a time. We can bump the API verison at some point to disallow these types of queries. In the long run we should be able to support all types of queries on the compute gateway and this API won't be needed anymore.

### CODE REVIEW

**Comment by kirankumarkolli** [comment]
This is the one CX would love, its huge. Thanks @bchong95 .

/cc: @mbhaskar , @moderakh , @kushagraThapar

File: Microsoft.Azure.Cosmos/src/Resource/Container/ContainerInternal.cs [hunk 1]
```diff
@@ -19,67 +19,67 @@ internal abstract class ContainerInternal : Container
     {
         public abstract Database Database { get; }
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Resource/Container/ContainerInternal.cs:20]: @j82w is ContainerInternal designed to scope if inside SDK only? If its planned to a contract for compute then below will open-up wider contract surface.

File: Microsoft.Azure.Cosmos/src/Resource/Container/ContainerCore.Items.cs [hunk 2]
```diff
@@ -321,6 +322,26 @@ internal partial class ContainerCore : ContainerInternal
                 throw new ArgumentNullException(nameof(requestOptions));
             }
 
+            if (feedRangeInternal != null)
+            {
+                // The user has scoped down to a physical partition or logical partition.
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Resource/Container/ContainerCore.Items.cs:327]: This assumption is not right. Single KeyRange can span across physical partitions.

File: Microsoft.Azure.Cosmos/src/Resource/Container/ContainerCore.Items.cs [hunk 3]
```diff
@@ -321,6 +322,26 @@ internal partial class ContainerCore : ContainerInternal
...
+                // In either case let the query execute as a passthrough.
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Resource/Container/ContainerCore.Items.cs:328]: Are all queries executed as pass trough? Thought aggregated still can't be done no?

File: Microsoft.Azure.Cosmos/src/Resource/Container/ContainerCore.Items.cs [hunk 4]
```diff
@@ -531,7 +556,7 @@ public override TransactionalBatch CreateTransactionalBatch(PartitionKey partiti
             return new BatchCore(this, partitionKey);
         }
 
-        internal override async Task<IEnumerable<string>> GetChangeFeedTokensAsync(CancellationToken cancellationToken = default(CancellationToken))
+        public override async Task<IEnumerable<string>> GetChangeFeedTokensAsync(CancellationToken cancellationToken = default(CancellationToken))
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Resource/Container/ContainerCore.Items.cs:559]: @ealsur in what scenarios is this used?

**Comment by kirankumarkolli** [comment]
Design discussion concluded it needs dependency on query parser, is that still the plan?

File: Microsoft.Azure.Cosmos/src/Resource/Container/ContainerInternal.cs (hunk 1 above)
Comment (j82w) [reply Microsoft.Azure.Cosmos/src/Resource/Container/ContainerInternal.cs:20]: The goal was ContainerInternal.Public can be used by Compute, but ContianerInternal.Internal is SDK only. We shouldn't make all these public it exposes to much of the API.

Comment (bchong95) [reply Microsoft.Azure.Cosmos/src/Resource/Container/ContainerInternal.cs:20]: It doesn't expose any of the API. The class is internal so the method is still internal. You should set the accessibility of the method as if the class were public. I don't get why we keep having this discussion. If you want to create a compute specific API set then that is up to you.

File: Microsoft.Azure.Cosmos/src/Resource/Container/ContainerCore.Items.cs (hunk 2 above)
Comment (bchong95) [reply Microsoft.Azure.Cosmos/src/Resource/Container/ContainerCore.Items.cs:327]: The API doesn't really have a nice way of checking if it's only one physical partition. I can try and write a visitor or enumerate the cases. For now the caller is never going to set it unless it's one of those two cases.

File: Microsoft.Azure.Cosmos/src/Resource/Container/ContainerCore.Items.cs (hunk 3 above)
Comment (bchong95) [reply Microsoft.Azure.Cosmos/src/Resource/Container/ContainerCore.Items.cs:328]: The PR description explains that is for compute to support the old legacy gateway contract.

**Comment by bchong95** [comment]
This PR is not for the general passthrough query approach we are going to provide for our customers in the future. This is to unblock compute on a legacy gateway contract support.

File: Microsoft.Azure.Cosmos/src/Resource/Container/ContainerCore.Items.cs (hunk 2 above)
Comment (bchong95) [reply Microsoft.Azure.Cosmos/src/Resource/Container/ContainerCore.Items.cs:327]: We can assert the count.

File: Microsoft.Azure.Cosmos/src/Resource/Container/ContainerCore.Items.cs (hunk 4 above)
Comment (bchong95) [reply Microsoft.Azure.Cosmos/src/Resource/Container/ContainerCore.Items.cs:559]: it's for the initial creation of parallel changefeed tokens on different machines

**Pull request merged**